### PR TITLE
Global style mixing

### LINKED ISSUE #59: Global style mixin

### PR DESCRIPTION
**Pull request by ibockowsky**
Close #59

### CODE REVIEW

File: src/utils/styleVariables/helpers/calculateCssSize.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  'rem',
+  'vh',
+  'vmin',
+  'vmax',
+  '%',
+  'auto'
+]
+
+const parseSizeWithUnit = (size) => {
+  if (isNaN(size) == false) return size + 'px'
```
Comment (allozaur) [inline src/utils/styleVariables/helpers/calculateCssSize.js:20]:
```suggestion
  if (isNaN(size) === false) return size + 'px'
```

File: src/components/nButton/nButton.scss [hunk 2]
```diff
@@ -21,9 +21,9 @@
   .n-button__text,
   .n-button__icon {
     @include flex($place: center);
-    color: var(--button-text-color);
+    color: var(--text-color);
     svg {
-      fill: var(--button-text-color) !important;
+      fill: var(--text-color) !important;
```
Comment (allozaur) [inline src/components/nButton/nButton.scss:26]: I'd rename this to CSS Native attribute name — `color`

File: src/components/nButton/nButton.vue [hunk 3]
```diff
@@ -31,12 +31,21 @@
 </template>
 
 <script>
-import calculateCssSizeMixin from '../../utils/calculateCssSize'
 import hrefIsExternalMixin from '../../utils/hrefIsExternal'
+import styleVariables from '../../utils/styleVariables'
+import { textColor, backgroundColor, padding, height, gap } from '../../utils/styleVariables/helpers/variables'
+
+const defaultStyleVariables = [
+  textColor,
```
Comment (allozaur) [inline src/components/nButton/nButton.vue:39]: `textColor` -> `color`

File: src/utils/styleVariables/helpers/variables.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+const color = {
+    name: 'color',
+    type: 'color'
+}
+
+const background = {
+    name: 'background',
+    type: 'color'
```
Comment (allozaur) [inline src/utils/styleVariables/helpers/variables.js:80]: `background` property expresses more than just a color, so i would treat it differently, maybe `type`: `shorthand`?

File: src/utils/styleVariables/helpers/variables.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    fontSize,
+    color,
+    background,
+    backgroundColor,
+    textColor,
+    spacing,
+    itemsGap,
+    viewportPadding,
+    resize
+}
```
Comment (allozaur) [inline src/utils/styleVariables/helpers/variables.js:112]:
```suggestion
}

```

File: src/utils/styleVariables/index.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    style () {
+      const defaultVariables = fillStylingObject(this.$props, defaultStyleVariables)
+      const customVariables = fillStylingObject(this.$props, customStyleVariables)
+
+      return {...defaultVariables, customVariables}
+    }
+  }
+})
+
+export default styleVariables
```
Comment (allozaur) [inline src/utils/styleVariables/index.js:35]:
```suggestion
export default styleVariables

```

**Review by allozaur — CHANGES_REQUESTED** [review]

File: src/utils/styleVariables/helpers/variables.js (hunk 4 above)
Comment (ibockowsky) [reply src/utils/styleVariables/helpers/variables.js:80]: Currently, for properties like `background` I've created `other` type

Comment (ibockowsky) [reply src/utils/styleVariables/helpers/variables.js:80]:
```suggestion
    type: 'other'
```

**Review by allozaur — APPROVED** [review]

**Pull request merged**